external/llvm-project/mlir/lib/Dialect/XModel/Transforms/SelectTargets.cpp

```diff
@@ -0,0 +1,95 @@
+//===- SelectTargets.cpp - Select IP target for execution -----------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements the async.launch pattern rewriter that converts kernel
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Async/IR/Async.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/XModel/IR/XModel.h"
+#include "mlir/Dialect/XModel/Transforms/Passes.h"
+#include "mlir/ExecutionEngine/SystemDevices.h"
+#include "mlir/IR/BlockAndValueMapping.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/SymbolTable.h"
+#include "mlir/Support/LLVM.h"
+#include "mlir/Transforms/RegionUtils.h"
+#include "llvm/ADT/TypeSwitch.h"
+
+namespace mlir {
+namespace xmodel {
+#define GEN_PASS_DEF_XMODELSELECTTARGETSPASS
+#include "mlir/Dialect/XModel/Transforms/Passes.h.inc"
+} // namespace xmodel
+} // namespace mlir
+
+#define DEBUG_TYPE "xmodel-select-targets"
+
+using namespace mlir;
+namespace {
+struct XModelSelectTargetsPass
+    : public xmodel::impl::XModelSelectTargetsPassBase<
+          XModelSelectTargetsPass> {
+  using xmodel::impl::XModelSelectTargetsPassBase<
+      XModelSelectTargetsPass>::XModelSelectTargetsPassBase;
+
+  bool testType(xmodel::TargetType type) const {
+    auto typeStr = xmodel::getNameForTargetType(type);
+    for (auto targetType : targetTypes) {
+      if (targetType == typeStr)
+        return true;
+    }
+    return false;
+  }
+
+  bool testArch(xmodel::TargetType type, StringRef arch) const {
+    if (!testType(type))
+      return false;
+    SystemDevice testDev{SystemDevice::Type::EGPU};
```

Comment:
Why is this hardcoded to GPU?

Reply:
The check is basically only comparing arch vs arch strings. So the types just need to be the same.